pkg/inspector/image-inspector.go

```diff
@@ -127,36 +133,76 @@ func (i *defaultImageInspector) Inspect() error {
 
 	ctx := context.Background()
 
-	imageMetaBefore, inspectErrBefore := client.InspectImage(i.opts.Image)
-	if i.opts.PullPolicy == iiapi.PullNever && inspectErrBefore != nil {
-		return fmt.Errorf("Image %s is not available and pull-policy %s doesn't allow pulling",
-			i.opts.Image, i.opts.PullPolicy)
-	}
+	if len(i.opts.Container) == 0 {
+		imageMetaBefore, inspectErrBefore := client.InspectImage(i.opts.Image)
+		if i.opts.PullPolicy == iiapi.PullNever && inspectErrBefore != nil {
+			return fmt.Errorf("Image %s is not available and pull-policy %s doesn't allow pulling",
+				i.opts.Image, i.opts.PullPolicy)
+		}
 
-	if i.opts.PullPolicy == iiapi.PullAlways ||
-		(i.opts.PullPolicy == iiapi.PullIfNotPresent && inspectErrBefore != nil) {
-		if err = i.pullImage(client); err != nil {
+		if i.opts.PullPolicy == iiapi.PullAlways ||
+			(i.opts.PullPolicy == iiapi.PullIfNotPresent && inspectErrBefore != nil) {
+			if err = i.pullImage(client); err != nil {
+				return err
+			}
+		}
+
+		imageMetaAfter, inspectErrAfter := client.InspectImage(i.opts.Image)
+		if inspectErrBefore == nil && inspectErrAfter == nil &&
+			imageMetaBefore.ID == imageMetaAfter.ID {
+			log.Printf("Image %s was already available", i.opts.Image)
+		}
+
+		randomName, err := generateRandomName()
+		if err != nil {
 			return err
 		}
-	}
 
-	imageMetaAfter, inspectErrAfter := client.InspectImage(i.opts.Image)
-	if inspectErrBefore == nil && inspectErrAfter == nil &&
-		imageMetaBefore.ID == imageMetaAfter.ID {
-		log.Printf("Image %s was already available", i.opts.Image)
-	}
+		imageMetadata, err := i.createAndExtractImage(client, randomName)
+		if err != nil {
+			return err
+		}
+		i.meta.Image = *imageMetadata
+		scanResults.ImageID = i.meta.Image.ID
+	} else {
+		meta, err := i.getContainerMeta(client)
+		if err != nil {
+			return err
+		}
 
-	randomName, err := generateRandomName()
-	if err != nil {
-		return err
-	}
+		i.meta.Image = *meta.Image
+		scanResults.ImageID = meta.Image.ID
+		scanResults.ContainerID = meta.Container.ID
 
-	imageMetadata, err := i.createAndExtractImage(client, randomName)
-	if err != nil {
-		return err
+		var filterInclude map[string]struct{}
```

Comment:
wouldn't this be better as a `map[string]bool`?

Reply:
@ilackarms I view them as basically equivalent (assuming that values are only ever set to true), but the map[string]struct{} uses less space. I've seen the empty struct idiom used frequently when a string set is wanted. This thread talks about it: https://groups.google.com/forum/#!topic/golang-nuts/lb4xLHq7wug

Reply:
personal preference: I prefer a little better readability over a little better performance